router/core/graphql_prehandler.go

```diff
@@ -597,6 +600,15 @@ func (h *PreHandler) handleOperation(req *http.Request, buf *bytes.Buffer, httpO
 
 	httpOperation.traceTimings.EndPlanning()
 
+	/**
+	* Validate that the planned query doesn't exceed the maximum query depth configured
+	 */
+
+	queryDepthErr := operationKit.ValidateQueryDepth(h.maxQueryDepth, opContext.preparedPlan.operationDocument, opContext.preparedPlan.schemaDocument)
```

Comment:
Depth analysis should be done before we planning to prevent attacks and computation.

Reply:
@StarpTech how can I get the operation/query document before the plan? Currently I'm getting the `*ast.Documents` necessary from the `opContext`

Reply:
You would need to parse it before https://github.com/wundergraph/cosmo/blob/41cbcad7e28b500b9ad08d92d3327694a579bd4c/router/core/operation_planner.go#L66 and the routerSchema is available globally.

Reply:
no, it's already parsed
take a look at how normalization or other validation steps use the operation document

Reply:
Right, it is already populated in `operationKit.kit.doc` after parsing.